### DISCUSSION TITLE
ability to only send `grab-keyboard` for a left or right meta key?

### DISCUSSION THREAD

**Discussion by ipatch** [FreeRDP/FreeRDP#8823] [Q&A]
recently started using xfreerdp to connect to a remote windows 7 machine. locally i'm using i3 on a arch linux. i use a keychron k6 keyboard with macos key caps. i have a command <kbd>⌘</kbd> key to the left and right of the space bar on my keybaord. i use the `-grab-keyboard` flag to allow me to use the _super_ / _meta_ key for my i3 window bindings. however is it possible use the left <kbd>⌘</kbd> for i3 related window bindings, but send the keyboard events to the rdp client if i press the **right** <kbd>⌘</kbd>. i use a bit of <kbd>⌘</kbd>+<kbd>r</kbd> binding (one example) when working with windows not being able to use the quote on quote _windows key_ is bit of a pain but not absolute deal breaker.

i'd find it hard to believe that i'd be the first person to ask this question.

**Comment by akallabeth** [comment]
there are
```
/kbd-list                         List keyboard layouts
    /kbd-lang-list                    List keyboard languages
    /kbd-remap:List of <key>=<value>,... pairs to remap scancodes
```

options for `xfreerdp`, so you can check if 
1. your layout may be supported (on the remote end)
2. you can rearrange keys to your liking with `/kbd-remap:` (only thing I don´t remember from top of my head is how to get the scancode)

**Reply by ipatch** [reply]
thank you for the prompt reply, i'll mess around with this in a bit. and if i get something going i'll post back make sure to mark your answer the solution.

thanks again. 🙏

**Reply by ipatch** [reply]
#8838

**Reply by ipatch** [reply]
pull request #7283 is definitely related

**Reply by ipatch** [reply]
> there are
> 
> ```
> /kbd-list                         List keyboard layouts
>     /kbd-lang-list                    List keyboard languages
>     /kbd-remap:List of <key>=<value>,... pairs to remap scancodes
> ```
> 
> options for `xfreerdp`, so you can check if
> 
>     1. your layout may be supported (on the remote end)
> 
>     2. you can rearrange keys to your liking with `/kbd-remap:` (only thing I don´t remember from top of my head is how to get the scancode)


if building from git master i'm able to list the scancodes with the below, still unable to send super left or super right independently from what i've been trying unfortunately. 

```
xfreerdp /list:kbd-scancode
```

**Reply by akallabeth** [reply]
@ipatch I did read your initial question wrong. No, you can only use `+grab-keyboard` or `-grab-keyboard` but not a mixture of it. So you can´t have one key captured by your local window manager and the other by the remote one.

The current escape is to move the mouse out of the session window to release the keyboard grab and then press the button.
for everything else you need to patch the `xfreerdp` client source code (check `xf_keyboard.c`  and maybe `xf_event.c` for that)
